### PR TITLE
Refactors library files and adds tests.

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
Follow up to this is going to be:

Refactor Setup
Refactor Vagrantfile
Add new hotness

File: lib/katello_deploy/repo_maker.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  class RepoMaker
+
+    attr_reader :name
+
+    def initialize(args)
+      @name = args.fetch(:name)
+      @directory = args.fetch(:directory)
+      @repo_file = RepoFile.new(
+        :name     => @name,
+        :baseurl  => File.expand_path(@directory)
```
Comment (stbenjam) [inline lib/katello_deploy/repo_maker.rb]: This needs to be prefixed by `file://`

**Comment by stbenjam** [comment]
>  Refactor Vagrantfile

Is that accidentally missing or still a WIP?

**Comment by stbenjam** [comment]
Tried:

```
./setup.rb --koji-repos --koji-task 256962 --module-prs gutterball/10
```

Seems to still work the same, except for the minor repo problem I mentioned.

Nice! :smiley:

**Comment by ehelms** [comment]
I should have been clearer on my intentions. The three items I mentioned I was planning to do as follow-up PRs  instead of one monolithic refactor and in that order: setup.rb, Vagrantfile, new stuff.

**Comment by ehelms** [comment]
Also, good catch on that naming issue, that led to needing to fully qualify paths in the module-prs portion as well which I have updated. Working locally for me, so please re-test.

**Comment by stbenjam** [comment]
Works great! ACK

**Comment by jlsherrill** [comment]
ACK from me

**Pull request merged**